src/Application/Console/Analyser.php

```diff
@@ -52,6 +52,10 @@ public function analyse(
         $insightCollection = $this->insightCollectionFactory
             ->get($metrics, $config, $dir, $consoleOutput);
 
+        if (method_exists($formatter, 'injectFileLinkFormatter')) {
```

Comment:
Not sure if I like this approach. I feel like it doesn't belong here. Could we just give the format method the config file or the config file in the constructor of the formatter?

Reply:
Yeah, I'm not convinced either. 

I initially added it in format function, but it's not a prerequisite for each formatter, that's why I think adding it in the constructor it's not a good solution... 

Maybe by create an Interface `RequireFileLinkFormatter` ? And adding it if formatter is instance of this interface. 
WDYT @olivernybroe ?

Reply:
Hmm, still seems kinda weird as that is only needed for that formatter.

What if we in the constructor of the Console formatter just extract the parameter `ide` from the `inputInteface`? This way all the logic is in the `Console` formatter, but no extra parameter is needed.

Reply:
In fact, it's used by the Console formatter, but we can imagine use it also in an HTML Formatter. That's why I would prefer keeping the ide resolving outside the Console Formatter.

Maybe the easier would be to create an Config class, that will be instancied when the app boots, and dispatch it through the container ? Then each class that need a part of config can get it from container ?

Reply:
I like that idea. That would work great for when we add more features. 

Another idea could be to use a trait, but I'll let you choose, I like both ideas.